components/script/dom/canvasrenderingcontext2d.rs

```diff
@@ -119,15 +114,32 @@ impl CanvasContextState {
     }
 }
 
+#[must_root]
 #[derive(JSTraceable, MallocSizeOf)]
 pub struct CanvasState {
     #[ignore_malloc_size_of = "Defined in ipc-channel"]
     ipc_renderer: IpcSender<CanvasMsg>,
     canvas_id: CanvasId,
+    state: DomRefCell<CanvasContextState>,
+    origin_clean: Cell<bool>,
+    canvas: Option<&HTMLCanvasElement>,
```

Comment:
Have callers pass in a `Option<&HTMLCanvasElement>` argument when necessary instead of storing this.